### PR TITLE
Merge beta to main

### PR DESCRIPTION
**Pull request by DEVTomatoCake**
- Add user page which visualizes stored user data
- General site updates, including adding translations to command descriptions
- Move code which was repeated on most pages to a single js file
- Add dropdown for login/logout/user dashboard

@FuntyGithub <3

### CODE REVIEW

**Review by FuntyGithub — APPROVED** [review]
also ich finds super .-.

File: assets/js/script.js [hunk 1]
```diff
@@ -52,6 +54,18 @@ function pageLoad() {
 		setTimeout(() => fadeIn(document.getElementById('cookie-container')), 1000);
 	};
 
+	const username = getCookie("user");
+	if (username) {
+		if (isMainPage) document.getElementById("username-content").innerHTML = "Hallo, <span class='accent'>" + username + "</span>!";
+		document.getElementById("username-header").innerText = username;
+		document.getElementsByClassName("accdropdown-content")[0].innerHTML = '<a href="/logout" translation="global.logout">Abmelden</a><a href="/dashboard/user">Eigene Daten ansehen</a>';
```
Comment (booky10) [inline assets/js/script.js:61]: Shouldn't this be translated?

File: dashboard/index.html [hunk 2]
```diff
@@ -107,10 +107,10 @@
 							document.getElementById("linksidebar").innerHTML =
 								'<div class="section middle"><p class="title">EINSTELLUNGEN</p>' +
 								"<br><button onclick='saveSettings();' class='save'><ion-icon name='save-outline'></ion-icon> SPEICHERN</button>" +
```
Comment (booky10) [inline dashboard/index.html:109]: Translations (same for the reactionrole dashboard)?

File: dashboard/index.html [hunk 3]
```diff
@@ -289,13 +284,10 @@ <h1>Die Server werden geladen...</h1>
...
 
-						let query = "";
-						if (params.has("guild")) query = "?guild=" + params.get("guild");
-
-						location.href = "https://" + location.host + "/dashboard" + query;
+						location.href = "https://" + location.host + "/dashboard" + (params.has("guild") ? "?guild=" + encode(params.get("guild")) : "");;
 					} else {
 						document.getElementById("root-container").innerHTML =
-							'<h1>Es gab einen Fehler beim Verarbeiten der API-Abfrage!</h1>' +
+							'<h1>Es gab einen Fehler beim Verarbeiten der API-Abfrage!</h1><br>' +
```
Comment (booky10) [inline dashboard/index.html:290]: Translations (also all the other internal server errors)?

File: dashboard/reactionroles/index.html [hunk 4]
```diff
@@ -314,7 +306,7 @@ <h1>Die Server werden geladen...</h1>
 
 			if (params.has('guild')) {
 				document.getElementById('root-container').innerHTML = '<h1>Die Reactionroles werden geladen...</h1>';
-				connectWS(params.get("guild"));
+				connectWS(encode(params.get("guild")));
 			} else if (params.has("guild_id") && getCookie("token")) {
 				document.getElementById("root-container").innerHTML = '<h1>Die Einstellungen werden geladen...</h1>';
```
Comment (booky10) [inline dashboard/reactionroles/index.html:311]: Translations?

File: dashboard/user/index.html [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+					</div>
+					<div class="accdropdown-content">
+						<a href="/login" translation="global.login"></a>
+					</div>
+				</div>
+			</div>
+
+			<div class="content" id="content">
+				<div id="root-container" class="row">
+					<h1>Deine Daten werden geladen...</h1>
```
Comment (booky10) [inline dashboard/user/index.html:144]: TrAnSlAtiOns (same for below script)?

File: index.html [hunk 6]
```diff
@@ -129,14 +130,19 @@
...
+					</div>
+					<div class="accdropdown-content">
+						<a href="/login" translation="global.login"></a>
+					</div>
 				</div>
 			</div>
 
 			<div class="content" id="content">
-				<h1 id="username-content" class="greeting" translation="main.hello">Hallo!</h1>
+				<h1 id="username-content" class="greeting">Hallo!</h1>
```
Comment (booky10) [inline index.html:145]: Translations?

**Review by booky10 — COMMENTED** [review]
I don't know if some texts are intentionally not translated, but I've marked them anyway.

File: assets/js/script.js (hunk 1 above)
Comment (DEVTomatoCake) [reply assets/js/script.js:61]: Translations coming soonTM in another update

**Pull request merged**